ginivision/src/doc/source/updating-to-2-4-0.rst

```diff
@@ -0,0 +1,445 @@
+
+.. code-block:: java
+
+    // As an example we show how to allow only jpegs and pdfs smaller than 5MB
+    @Override
+    public void onCheckImportedDocument(@NonNull final Document document,
+            @NonNull final DocumentCheckResultCallback callback) {
+        // We can apply custom checks here to an imported document and notify the Gini Vision
+        // Library about the result
+        // IMPORTANT: do not call super as it will lead to unexpected behaviors
+
+        if (DO_CUSTOM_DOCUMENT_CHECK) {
+            // Use the Intent with which the document was imported to access its contents
+            // (document.getData() may be null)
+            final Intent intent = document.getIntent();
+            if (intent == null) {
+                callback.documentRejected(getString(R.string.gv_document_import_error));
+                return;
+            }
+            final Uri uri = IntentHelper.getUri(intent);
+            if (uri == null) {
+                callback.documentRejected(getString(R.string.gv_document_import_error));
+                return;
+            }
+            if (hasMoreThan5MB(uri)) {
+                callback.documentRejected(getString(R.string.document_size_too_large));
+                return;
+            }
+            // IMPORTANT: always call one of the callback methods
+            if (isJpegOrPdf(uri)) {
+                callback.documentAccepted();
+            } else {
+                callback.documentRejected(getString(R.string.unsupported_document_type));
+            }
+        } else {
+            // IMPORTANT: always call one of the callback methods
+            callback.documentAccepted();
+        }
+    }
+
+Rejecting a document displays the provided message to the user in an alert dialog. The user may select another document or cancel the document import.
+
+.. _Document Import - Analyzing Imported Documents:
+
+Analyzing Imported Documents
+~~~~
+
+After the document was accepted the Gini Vision Library shows the Review Screen for images and for PDFs it goes directly to the Analysis Screen. Analyzing imported documents requires no changes. The document's content will be loaded into memory and can be uploaded like the pictures taken by the camera.
+
+Additional methods were added to the ``Document`` to identify whether the document was imported, to get the ``Intent`` with which it was imported, to find out if it's reviewable (PDFs are not reviewable for ex.) and to get it's type (image or pdf).
+
+Use the additional methods in the Review and Analysis screens if you wish to handle imported documents separately from pictures taken by the camera.
+
+.. _Document Import - Analysis Screen for PDFs:
+
+Analysis Screen for PDFs
+~~~~
+
+The Analysis Screen will render the first page of the PDF on Android Lollipop and newer. On older versions no preview is shown. In addition above the PDF preview area the PDF filename is displayed and the nr of pages (on Android Lollipop and newer).
```

Comment:
Can you add a version number to Lollipop, please? It makes it easier for me to communicate with the developers ...